walletrpc/service.proto

```diff
@@ -118,6 +118,23 @@ message TreeState {
     string orchardTree = 6; // orchard commitment tree state
 }
 
+enum ShieldedProtocol {
+    SAPLING = 0;
+    ORCHARD = 1;
+}
+
+message GetSubtreeRootsArg {
+    uint64 startIndex = 1;                  // Index identifying where to start returning subtree roots
+    ShieldedProtocol shieldedProtocol = 2;  // Shielded protocol to return subtree roots for
+    uint64 limit = 3;                       // Maximum number of entries to return, or 0 for all entries.
```

Comment:
```suggestion
    uint64 maxEntries = 3;                  // Maximum number of entries to return, or 0 for all entries.
```
I suggest using this name because we use it for another gRPC (and it has the same meaning), and I think it's slightly more descriptive, because some may interpret "limit" as a stopping index.